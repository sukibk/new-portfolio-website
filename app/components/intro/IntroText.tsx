import { TypeAnimation } from "react-type-animation";

/**
 * IntroText Component
 *
 * Displays a typing animation introducing the developer.
 *
 * @component
 * @returns JSX.Element
 *
 * @example
 * <IntroText/>
 */
const IntroText = () => (
  <TypeAnimation
    className="h-[8rem] p-4 !text-foreground-title lg:max-w-1/2 font-light"
    sequence={[
      "Hi!",
      1000,
      "My name is Marko Sudar",
      2000,
      "I have been developing apps for a couple of years now",
      2000,
      "I have worked as Cloud Engineer and Web developer at Oracle, " +
        "and Software Engineer at AUVSI",
      2000,
      "Graduated with 4.0 GPA from Catawba College in CS Major",
      2000,
      "Welcome to my portfolio!",
    ]}
    wrapper="p"
    speed={50}
    deletionSpeed={90}
    cursor
    repeat={0}
    style={{
      fontSize: "2em",
      display: "inline-block",
      color: "white",
    }}
  />
);

export default IntroText;
