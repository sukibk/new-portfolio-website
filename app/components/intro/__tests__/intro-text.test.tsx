import { render } from "@testing-library/react";

import IntroText from "../IntroText";

describe("IntroText", () => {
  it("renders without crashing", () => {
    const { container } = render(<IntroText />);
    expect(container).toMatchSnapshot();
  });
});
