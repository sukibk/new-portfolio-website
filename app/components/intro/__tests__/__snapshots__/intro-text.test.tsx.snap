// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IntroText > renders without crashing 1`] = `
<div>
  <p
    class="index-module_type__E-SaG h-[8rem] p-4 !text-foreground-title lg:max-w-1/2
    font-light font-code"
    style="font-size: 2em; display: inline-block; color: white;"
  />
</div>
`;
